format/src/matbin/mod.rs

```diff
@@ -0,0 +1,309 @@
+use std::{borrow::Cow, io};
+
+use bytemuck::PodCastError;
+use byteorder::LE;
+use thiserror::Error;
+use widestring::{U16Str, U16String};
+use zerocopy::{FromBytes, FromZeroes, Ref, F32, U32, U64};
+
+use crate::io_ext::zerocopy::Padding;
+
+#[derive(Debug, Error)]
+pub enum MatbinError {
+    #[error("Could not copy bytes {0}")]
+    IO(#[from] io::Error),
+
+    #[error("Could not read string")]
+    String(#[from] ReadUtf16StringError),
+
+    #[error("Got unknown parameter type {0}")]
+    UnknownParameterType(u32),
+
+    #[error("Could not create reference to value")]
+    ValueZerocopy,
```

Comment:
The only time this error can be produced (if bounds are already checked) is when the data is unaligned.

Reply:
Renamed to `UnalignedValue` in f8ecf02d3d602cce65538df48def9046dd3255fc